pos_picking_delayed/models/pos_order.py

```diff
@@ -0,0 +1,57 @@
+# coding: utf-8
+# Copyright 2018 - Today Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    has_picking_delayed = fields.Boolean(
+        default=False, help="This checkbox is checked if the generation of"
+        " the picking has been delayed. The picking will be created by cron.")
+
+    # Overload Section
+    @api.model
+    def create_from_ui(self, orders):
+        PosSession = self.env['pos.session']
+        for order_data in orders:
+            session_id = order_data.get('data').get('pos_session_id')
+            session = PosSession.browse(session_id)
+            order_data['data']['has_picking_delayed'] =\
+                session.config_id.picking_creation_delayed
+        return super(PosOrder, self.with_context(
+            create_from_ui=True)).create_from_ui(orders)
+
+    def create_picking(self):
+        if self.env.context.get('create_from_ui', False):
+            orders = self.filtered(lambda x: not x.has_picking_delayed)
+        else:
+            orders = self
+        res = super(PosOrder, orders).create_picking()
+        orders.write({'has_picking_delayed': False})
+        return res
+
+    @api.model
+    def _order_fields(self, ui_order):
+        res = super(PosOrder, self)._order_fields(ui_order)
+        res['has_picking_delayed'] = ui_order['has_picking_delayed']
+        return res
+
+    # Custom Section
+    @api.model
+    def create_delayed_picking(self):
+        orders = self.search([
+            ('state', '!=', 'draft'),
+            ('has_picking_delayed', '=', True),
+        ], order='date_order')
+        for order in orders:
+            order.sudo(order.user_id.id).with_context(
```

Comment:
Even if this is called through cron, you will create two environments (one for 'sudo' and one for 'with_context').

Why don't you use queue_job process ? Moreover, it will allow you to follow the picking creation status in job lines.

Reply:
> Even if this is called through cron, you will create two environments (one for 'sudo' and one for 'with_context').

Do you have an alternative ?

> Why don't you use queue_job process ? Moreover, it will allow you to follow the picking creation status in job lines.

Indeed, could you be a great improvement. I personaly don't use queue_job, but feel free to make a PR to improve this module ! Otherwise, we can add a simple remark in the roadmap section.

Reply:
> > Even if this is called through cron, you will create two environments (one for 'sudo' and one for 'with_context').
> 
> Do you have an alternative ?

One could be grouping orders by company and changing context once.

> 
> > Why don't you use queue_job process ? Moreover, it will allow you to follow the picking creation status in job lines.
> 
> Indeed, could you be a great improvement. I personaly don't use queue_job, but feel free to make a PR to improve this module ! Otherwise, we can add a simple remark in the roadmap section.

Ok for ROADMAP